### ISSUE TITLE
Better ordering of search results

### ISSUE THREAD

**Issue by martinjoconnor** [metadatacenter/cedar-resource-server#81]
Currently, the non-field-based ngram-based search does not attempt to order the results. For example, if we search for the string 'ct' it can match artifacts with names or descriptions containing 'ct' anywhere in a word (case insensitively). 

e.g.,

![Screen Shot 2019-05-16 at 2.24.46 PM.png](https://images.zenhubusercontent.com/559ac6c34f7a08fbb7f3984b/84777a98-fee2-4b0e-8a90-59ed76730919)

It might be better to return full word matches for the search term first, for example. Also prioritizing names above descriptions in the search?

Other suggestions?

**Comment by graybeal** [comment]
Yes please, I just got quizzed on how BioPortal handles this today. Full word match highest priority, in name higher priority than description, beginning of field slightly higher priority than later in field. 

And later, when we actually can tell which templates/instances are most used, most visited, or most edited, use that as a proxy for a popularity priority to order the full word match cases.
